packages/components/Switch/src/useSwitch.ts

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { useAsPressable, useKeyProps, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { SwitchProps, SwitchInfo } from './Switch.types';
+import { memoize } from '@fluentui-react-native/framework';
+import { AccessibilityState, AccessibilityActionEvent, LayoutAnimation, AccessibilityActionInfo } from 'react-native';
+import { InteractionEvent } from '@fluentui-react-native/interactive-hooks';
+
+export const useSwitch = (props: SwitchProps): SwitchInfo => {
+  const defaultComponentRef = React.useRef(null);
+  const {
+    onChange,
+    checked,
+    defaultChecked,
```

Comment:
Are the `defaultChecked` and `checked` props mutually exclusive or can they both be set? What's the expected behavior if they're both set? 

I would think setting `defaultChecked` means you're setting the initial value for Switch and that it would be mutually exclusive to `checked`. Using `checked` would mean you're defining the checked state of Switch (so setting `defaultChecked` wouldn't matter) and you intend to control the value of checked at a higher level instead of letting the Switch manage the state.

If they are mutually exclusive, I would add a check that warns the user to use one or the other and not set both.

Reply:
They are mutually exclusive. I can add a check that warns users.